src/include/distributed/multi_physical_planner.h

```diff
@@ -169,6 +169,9 @@ typedef struct Task
 	uint64 shardId;                /* only applies to shard fetch tasks */
 	TaskExecution *taskExecution;  /* used by task tracker executor */
 	bool upsertQuery;              /* only applies to modify tasks */
+
+	bool insertSelectQuery;
```

Comment:
Let me first share my observations. On `RouterExecutablePlan ()`, I had to do a hack to allow zero-shard `INSERT ... SELECT` queries to be qualifed as router executable. I'm really not happy with that.

After seeing this `insertSelectQuery`, I'm thinking that it'd be great if we can move that to `workerJob` and then pass that information to required funtions here. And, I could be able to use that flag on `RouterExecutablePlan ()`.

I can see that it'd complicate this PR, but, ease the insert/select pr.

What do you think about moving it to `workerJob`?